src/builder/javascript.js

```diff
@@ -1,16 +1,29 @@
 import { transformFile } from 'babel-core'
 import { default as deferred } from 'thenify'
 import { default as compat } from 'babel-plugin-add-module-exports'
-import { default as es2015 } from 'babel-preset-es2015'
-import { default as umd } from 'babel-plugin-transform-es2015-modules-umd'
+import { default as ecmascript } from 'babel-preset-latest'
+
+const babelrc = process.env.NODE_ENV !== 'test'
 
 export default function configure(pkg, opts) {
   return async function process(name, file) {
-    let presets = [es2015]
-      , plugins = [umd]
+    const { es2017
+          , modules
+          , ['add-module-exports']: addModuleExports
+          } = opts.flags
+
+    let presets = [
```

Comment:
You're right, and this is something we've discussed in the past but not found a particularly elegant solution for. Part of the issue is that we want to not give out too much control, since [choice is not a feature](/zambezi/ez-build/rationale.md), but locking us out of presets and plugins is just to shoot ourselves in both feet at the same time.

Another issue is that if we say you can define the set of presets and plugins *as you see fit* it'd make it much harder for us to provide options like `--production` and not lie, since the optimization we'd want to do would almost certainly mean including a specific set of plugins and if setting a list of presets or plugins would effectively opt you out of that then our CLI options run the risk of becoming meaningless, and we fail to deliver a simple tool that takes away the complexity of configuration.

You're not wrong, but maybe it's something we should discuss in a separate issue? Would love to try and hash this out!